fe/fe-core/src/main/java/com/starrocks/qe/MasterOpExecutor.java

```diff
@@ -51,24 +49,22 @@ public class MasterOpExecutor {
     // the total time of thrift connectTime add readTime and writeTime
     private int thriftTimeoutMs;
 
-    private boolean shouldNotRetry;
-
-    public MasterOpExecutor(OriginStatement originStmt, ConnectContext ctx, RedirectStatus status, boolean isQuery) {
-        this(null, originStmt, ctx, status, isQuery);
+    public MasterOpExecutor(OriginStatement originStmt, ConnectContext ctx, RedirectStatus status) {
+        this(null, originStmt, ctx, status);
     }
 
     public MasterOpExecutor(StatementBase parsedStmt, OriginStatement originStmt,
-                            ConnectContext ctx, RedirectStatus status, boolean isQuery) {
+                            ConnectContext ctx, RedirectStatus status) {
         this.originStmt = originStmt;
         this.ctx = ctx;
         if (status.isNeedToWaitJournalSync()) {
             this.waitTimeoutMs = ctx.getSessionVariable().getQueryTimeoutS() * 1000;
         } else {
             this.waitTimeoutMs = 0;
         }
-        this.thriftTimeoutMs = ctx.getSessionVariable().getQueryTimeoutS() * 1000;
-        // if isQuery=false, we shouldn't retry twice when catch exception because of Idempotency
-        this.shouldNotRetry = !isQuery;
+        // set thriftTimeoutMs to 2 * query_timeout
```

Comment:
The query timeout user may set very short or very long. Therefore, it is best to configure net timeout independently of query timeout and extract it separately. thrift timeout is recommended to be set to query timeout + net timeout.

Reply:
done